### ISSUE TITLE
File dialogs (open, import, etc) does not offer Network or mapped network drives on Manjaro/Plasma

### ISSUE THREAD

**Issue by tped** [OpenShot/openshot-qt#3862]
[Referred over here from Reddit](https://www.reddit.com/r/OpenShot/comments/jxu64i/file_dialogs_open_import_etc_does_not_offer/)

**Question:**
The file dialog in OpenShot 2.5.1 on Manjaro/Plasma (arm64) does not offer Network shares or 'Remote Places' for open, import, save, etc. Samba is installed and functioning for other apps. Default file manager is Dolphin - I'd think a dolphin-like dialog would appear when I try to open files. OpenShot only offers local media, harddisk, removable storage.

My situation: I am moving to Manjaro/Plasma ARM64 on a Raspberry Pi 4 from Mint/Cinnamon in my x86 world and struggling to get Openshot 2.5.1 to work with projects and videos stored on our NAS.   ... and YES, OpenShot seems quite responsive and usable on the little Pi! 

Similar issue w/ latest Firefox and managed to fix using this article - [https://www.reddit.com/r/kde/comments/a5cxwk/firefox_v64_can_now_use_the_kde_file_selection/](https://www.reddit.com/r/kde/comments/a5cxwk/firefox_v64_can_now_use_the_kde_file_selection/). Have not found similar for OpenShot.

It sounds like Manjaro/ARM folk do ARM compiles, but I'd think this would be an issue on x86 also.  I'll post over there too and provide this link

**System Details:**
OpenShot Version 2.5.1 (from Manjaro official repository) 
Operating System: Manjaro-ARM
KDE Plasma Version: 5.20.3
KDE Frameworks Version: 5.76.0
Qt Version: 5.15.1
Kernel Version: 5.4.74-1-MANJARO-ARM
OS Type: 64-bit
Processors: 4 × 
Memory: 3.7 GiB of RAM
Graphics Processor: V3D 4.2

**Screenshots:** (Optional)
![netconfusion](https://user-images.githubusercontent.com/6204822/100102693-939cdd80-2e29-11eb-8723-52c34cecfb60.jpg)

**Comment by USATechDude** [comment]
I'm having the same issue on MX Linux 19.3.  When I run OpenShot from the command line, then open one of the File Open dialogs, this message appears on the console:

`Non-native QFileDialog supports only local files`

**Comment by ferdnyc** [comment]
@USATechDude If you're seeing messages about the _non_-native file dialog, then you're missing a platform plugin. OpenShot will use the native file dialogs for the current desktop environment if they're available.

Is MX a Debian derivative? I find Debian's packaging decisions on many things (Python modules, Qt components) to be incredibly bizarre and frustrating, because they seem to leave installation of non-required components as a thing the end user should somehow divine the need for. (...That sentence is a mess.)

Anyway, you can run `qtdiag` to get a profile of your Qt installation. On Fedora, platform components and other plugins are pulled in by default, so there will be a section like this:
```text
Theme:
  Platforms requested : gtk3,gnome,generic
            available : gtk2,gtk3,qt5ct,snap,flatpak,xdgdesktopportal
  Styles requested    : fusion,windows
         available    : Adwaita-Dark,Adwaita,Highcontrast,bb10dark,bb10bright,kvantum-dark,kvantum,cleanlooks,gtk2,cde,motif,plastique,qt5ct-style,Oxygen,QtCurve,Windows,Fusion
  Icon theme          : Adwaita,  from /home/ferd/.local/share/icons,/home/ferd/share/icons,/var/lib/flatpak/exports/share/icons,/usr/share/icons
  System font         : "Source Sans Pro" 11
  Native file dialog
  Native color dialog
  Native font dialog
```

The component that provides the native dialogs, since my DE is Gnome Shell, is `/usr/lib64/qt5/plugins/platformthemes/libqgtk3.so`. Installing whatever `apt` package provides `/usr/lib/x86_64-linux-gnu/qt5/plugins/platformthemes/libqgtk3.so` should enable the same thing on Debian-derived distros.

**Comment by ferdnyc** [comment]
Regarding the original report, some time after the 2.5.1 release (months after, in fact) I switched OpenShot's media import code over to using `QFileDialog.getOpenFileUrls()`, rather than `.getOpenFileNames()`, partly in the hopes that requesting the references as `QUrl`s would activate whatever support for network file sources is available natively on the various OSes and distros. So, this may be fixed in the Daily Build code — I'd certainly be interested to hear either confirmation on that (for each of the three OSes), or reports to the contrary, as I can try to look into additional steps that may be necessary.

But as Qt's console message says, it _definitely_ won't work unless native dialogs are enabled via the appropriate platform theme plugin.

**Comment by ferdnyc** [comment]
> On Fedora, platform components and other plugins are pulled in by default

Actually, that's a misstatement on my part. The Gtk3 platform plugin isn't merely "pulled in" by default. It's **PART OF** the `qt5-qtbase-gui` RPM, the one that provides `libQt5Gui.so.5` itself. So it's literally _not possible_ to have QtGui installed on Fedora, _without_ the platform plugin. 

Because packaging them all separately makes the system configuration as "flexible" as possible, sure, but it just makes things harder for the user by turning software installation into some kind of giant jigsaw puzzle they're forced to assemble themselves. (IMNSHO.)

**Comment by ferdnyc** [comment]
Certainly, in Fedora running 2.5.1-dev2, as you can see all supported sources, local and network filesystems alike, are accessible in the import dialog. But on Linux network filesystems still get mapped to local paths, so I believe this is unchanged from 2.5.1 release. The determining factor for Linux is having native dialogs enabled via the platform theme plugin. The QUrl change was more about Windows and MacOS.

![image](https://user-images.githubusercontent.com/538020/100155712-b7861080-2e75-11eb-8429-e79d7dc1554a.png)

**Comment by ferdnyc** [comment]
Finally (sorry!) there's also a _possibility_ that the AppImages don't have native dialog support. For the longest time there was nothing we could do about that, because we were packaging them with a very ancient Qt 5.2.1. Native dialog support for Gtk3 first appeared in Qt 5.7.

But as of September-ish, the Daily Builds use Qt 5.9. So if native dialogs still aren't working, we may just need to add the platform theme plugins to our AppImage packaging.

**Comment by tped** [comment]
Sorry about delay, it looked like you were looking for some feedback from each of us ... 

On Majaro (Arch), I installed openshot from main repository, not AppImage.  Still fumbling a bit with pacman, but it sure looks like I have platform plugins, I can see that libqgtk3.so has been installed:
```
$pacman -F libqgtk3.so
extra/qt5-base 5.15.1-3 (qt qt5) [installed]
    usr/lib/qt/plugins/platformthemes/libqgtk3.so
extra/qt5-es2-base 5.15.1-1 (qt qt5)
    usr/lib/qt/plugins/platformthemes/libqgtk3.so
```
Here also is theme part from qtdiag ... 
```
Theme:
  Platforms requested : kde,generic
            available : kde,gtk3,snap,flatpak,xdgdesktopportal
  Styles requested    : Breeze,breeze,oxygen,fusion,windows
         available    : Breeze,Windows,Fusion
  Icon theme          : breath2-dark, hicolor from /usr/share/icons
  System font         : "Noto Sans" 10
```
The [firefox 'fix'](https://www.reddit.com/r/kde/comments/a5cxwk/firefox_v64_can_now_use_the_kde_file_selection/) had me install xdg-desktop-portal and xdg-desktop-portal-kde packages, but I also had to force firefox to use it whenever I run the program like this:   `GTK_USE_PORTAL=1 /usr/lib/firefox/firefox %u`.  Seemed odd that I may have to deal with each app individually. 

I'll see if I can install OpenShot from an AppImage too - just to see if works and how performance is 

Thanks for looking into this - Let me know if you need any more info from my world

**Comment by ferdnyc** [comment]
@tped Er, yeah, sorry, lines sort of got crossed there. If you're running a _Plasma_ desktop, then your native environment is KDE, and you'd be looking for _those_ dialogs as your native ones. Switching to `gtk3` would get you the dialog you had with Firefox **before** you added those options to switch what it's using (the same one as in my screenshots).

But that Firefox change is necessary because Firefox is a Gtk application, natively. OpenShot is natively a Qt application, and KDE is based on Qt, so I'd think if anything native KDE dialogs should be available automatically. You're definitely not getting a Gtk dialog in OpenShot now, that much is clear from the screenshot you posted. And your `qtdiag` output seems to show `kde` as the preferred platform.

Were there any `Native Dialogs` lines below the `Theme` lines you posted? If not, that would definitely seem to be the issue, but I know too little about KDE to really guess what could be going on or how to solve it.

There is one simple test we can try, to determine what Qt is defaulting to. If you create a file `dialogtest.qml` anywhere, with these contents:
```qml
import QtQuick 2.3
import QtQuick.Dialogs 1.1

FileDialog

{Component.onCompleted: visible = true}
```
And then from a terminal in the same directory you saved that file, run:
```sh
qmlscene-qt5 dialogtest.qml
```

A file dialog should immediately open using whatever file dialog Qt is defaulting to. (You can just cancel it, you may have to hit <kbd>Ctrl</kbd>+<kbd>c</kbd> after to terminate the QML previewer.) I would _think_ what you get should be the KDE native dialog, based on most of the info you provided, but the lack of a `Native file dialog` line in your `qtdiag` output is troubling. And I have absolutely no idea how to go about fixing it, if so.

(But you _definitely_ don't need to care about `usr/lib/qt/plugins/platformthemes/libqgtk3.so`, that would just get you the Gtk native dialogs you don't want. If anything you'd want an equivalent for KDE, but the fact that `kde` is listed first in both your `Platforms requested:` and `available:` lists would seem to indicate that's already in use.)

**Comment by ferdnyc** [comment]
(What does your Firefox file-open dialog look like _after_ the change, out of curiosity? Presumably that's a goal we could aim for...)

**Comment by tped** [comment]
No prob, I really appreciate you looking into this.  I sure hope this isn't some config issue that should have been posted to Majaro or Arch guys in the first place!  I'll do more digging in forums based on what I've learned here from you.

Here are some answers for you

1) no mention of Native Dialogs in qtdiag ... I posted entire Theme Section.  That is troubling
2) Output from `qmlscene-qt5 dialogtest.qml` is below ... It sure looks like KDE native dialog
3) Attached is a screenshot of Firefox file-open after the mod also 

![qmlscene-output](https://user-images.githubusercontent.com/6204822/100872635-b5b8e000-3467-11eb-98fd-055c5de272cf.png)


![ffdialog-after](https://user-images.githubusercontent.com/6204822/100872713-d5e89f00-3467-11eb-9c2f-a9f7b65f8fcb.png)

**Comment by USATechDude** [comment]
@ferdnyc Sorry for the long delay, I underwent a major search and destroy mission trying to discover why my dialog boxes were different.  I finally rebuilt my MX Linux partition in favor of a Kubuntu 20.04 LTS distribution.  What worried me was that when I downloaded OpenShot and launched it, the dialog boxes were still the same as on my MX Linux system.  I gave in and downloaded the PPA of the Daily Build and was surprised when I found that the KDE native dialog boxes worked (with the exception of having network access) whereas the AppImage does not.

Why is there such a difference between the PPA and the AppImage?  Attached is a screenshot showing OpenShot AppImage on the left and the PPA on the right.  Is this issue unique to my system?
![Screenshot_20201202_124524](https://user-images.githubusercontent.com/65924242/100917194-722c9900-349c-11eb-9edb-eed272de0805.png)

**Comment by ferdnyc** [comment]
@USATechDude No, I suspect that's normal. The AppImages are running from their own bundled copy of Qt, and the native dialogs rely on certain platform plugins being installed on the system. But the AppImages will never see any plugins installed in the system Qt.

In theory the AppImage Qt _might_ be able to support native dialogs, if we bundled all of the various platform plugins in with its local Qt. But I'm not super confident about that because the native dialogs inherently require communication with the host system.

**Comment by USATechDude** [comment]
@ferdnyc OK, I think I understand.  I know that @jonoomph wants to reduce the binary file size, so adding more plugins is most likely not worth mentioning.

I'm still having the original issue that my dialog boxes don't have access to the network places.  Using all of the debug commands that I can find, and even a couple that @jonoomph gave me in a separate conversation, I can't find any error messages related to opening the dialog boxes now that I've rebuilt my system.  I'm starting to think that this is a "feature." :grin: 

Thank you for all of your help and suggestions. I appreciate the effort that you put forth in helping all of us.

**Comment by tped** [comment]
No prob, I really appreciate you looking into this.  I sure hope this isn't some config issue that should have been posted to Majaro or Arch guys in the first place!  I'll do more digging in forums based on what I've learned here from you.

Here are some answers for you

1) no mention of Native Dialogs in qtdiag ... I posted entire Theme Section.  That is troubling
2) Output from `qmlscene-qt5 dialogtest.qml` is below ... It sure looks like KDE native dialog
3) Attached is a screenshot of Firefox file-open after the mod also 

![qmlscene-output](https://user-images.githubusercontent.com/6204822/100872635-b5b8e000-3467-11eb-98fd-055c5de272cf.png)


![ffdialog-after](https://user-images.githubusercontent.com/6204822/100872713-d5e89f00-3467-11eb-9c2f-a9f7b65f8fcb.png)
